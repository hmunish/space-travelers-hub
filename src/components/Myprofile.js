import React from 'react';
import { useSelector } from 'react-redux';
import './profile.css';

const Myprofile = () => {
  const reservedRocketsNameList = useSelector(
    (state) => state.rocketsList.reservedRocketsName,
  );

  let missions = useSelector((state) => state.missions) || [];
  const savedMissions = JSON.parse(localStorage.getItem('missions')) || [];
  if (savedMissions.length > 0) {
    missions = savedMissions;
  }
  const myMissions = missions.filter((mission) => mission.joined === true);

  return (
    <section className="display-missions-rockets">
      <div>
        <h1>My missions</h1>
        <ul>
          {myMissions.map((mission) => (
            <li key={mission.id}>{mission.name}</li>

          ))}
        </ul>
      </div>
      <div>
        <h1>My Rockets</h1>
        <ul>
          {reservedRocketsNameList.map((el) => (
            <li key={el}>{el}</li>
          ))}
        </ul>
      </div>
    </section>
  );
};

export default Myprofile;
