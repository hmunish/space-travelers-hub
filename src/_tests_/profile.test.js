import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Myprofile from '../components/Myprofile';

const mockStore = configureStore([]);

describe('test rockets component', () => {
  const mockRockets = [
    {
      id: 1,
      name: 'Falcon',
      flickrImages: '',
      description: 'Falcon description',
      reserved: false,
    },
    {
      id: 2,
      name: 'Starship',
      flickrImages: '',
      description: 'Starship description',
      reserved: true,
    },
  ];

  const store = mockStore({
    rocketsList: {
      value: mockRockets,
      loading: false,
      errors: null,
      reservedRocketsName: [],
    },
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Myprofile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
