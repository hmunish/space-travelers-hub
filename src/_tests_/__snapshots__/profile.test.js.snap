// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test rockets component should match the snapshot 1`] = `
<section
  className="display-missions-rockets"
>
  <div>
    <h1>
      My missions
    </h1>
    <ul />
  </div>
  <div>
    <h1>
      My Rockets
    </h1>
    <ul />
  </div>
</section>
`;
